src/databricks/labs/ucx/source_code/files.py

```diff
@@ -138,3 +138,34 @@ def _apply_file_fix(self, path):
                 logger.info(f"Overwriting {path}")
                 f.write(code)
                 return True
+
+
+class LocalFileLinter:
+
+    def __init__(self, languages: Languages):
+        self._languages = languages
+        # TODO: Needs to support local notebooks, as well as pure python files
+        self._extensions = {".py": Language.PYTHON}
+
+    def lint(self, path: Path) -> bool:
+        if path.is_dir():
+            for child_path in path.iterdir():
+                self.lint(child_path)
+            return True
+        return self._lint_file(path)
+
+    def _lint_file(self, path: Path) -> bool:
+        if path.suffix not in self._extensions:
+            return False
+        language = self._extensions[path.suffix]
```

Comment:
initialize languages for every file, because we have current session state reset for every file

https://github.com/databrickslabs/ucx/blob/03f911a9b292aa350716bd0e906b77c0291e29cd/src/databricks/labs/ucx/source_code/languages.py#L12-L17